### DISCUSSION TITLE
TShock 4.5.2 (Terraria 1.4.2.2)

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#2283] [Show and tell]
_This release supports Terraria protocol 1.4.2.2._
## Notable changes
* Added preliminary support for Terraria 1.4.2.2. (@hakusaro)
* Removed `/ungodme` and godmode warning (no longer necessary). Also, godmode now supports silent commands. (@hakusaro)

## In other news
* Tonight (April 22) in North America is expected to be the peak of the [April Lyrids meteor shower](https://en.wikipedia.org/wiki/Lyrids). This happens every year, and is visible in most of the world. Why not look at the sky tonight, and make a wish?
* I feel quite shy, but I would like to say that @hakusaro can now be [sponsored on GitHub](https://github.com/hakusaro). I would like to formalize this a little bit. If you have opinions on these shoutouts for sponsorships, please drop them in this release thread.

<hr /><em>This discussion was created from the release <a href='https://github.com/Pryaxis/TShock/releases/tag/v4.5.2'>TShock 4.5.2 (Terraria 1.4.2.2)</a>.</em>

**Comment by hakusaro** [comment]
In the past, TShock accepted donations for server hosting and other things we needed to pay for. However, I shut down this program when it became clear that I could afford the hosting myself. That, and PayPal was really risky to support.

Now, though, GitHub sponsors exists. The difference is that TShock/Pryaxis will not accept donations, but many of our developers do. I've been encouraging people to setup sponsors profiles and pay them, because I feel like they deserve it for working on the project. Today is the first time I've admitted that I also have a sponsors profile. However, this is separate from a TShock donation: it's more about me directly working on TShock. This is separate for each developer, including me, for a reason. That way, you can sort of say "I like this developer's work specifically" instead of guessing where the money would go if it was the entire organization.

What I would like to do is change our GitHub sponsors button to list developers who have recently contributed to TShock, and have sponsors setup. DeathCradle is the only person who only accepts Bitcoin right now, so we'd have to figure something out for him. Then, we'd indicate that these developers have sponsors in each release with a small note next to their change.

The catch is that we'd need to remove developers from our GitHub sponsors button, if they stop contributing to TShock. My thinking is that we can add contributors we know, and remove them if they go inactive for a few months, give or take. The issue is that it's a bit arbitrary who we would add to the button, because we want to avoid one-off contributors, and we also don't want to burn people who contribute a lot intermittently.

Right now, I suspect that people don't often donate to GiveDirectly. So, I think the sponsors button might be more effective if it were directly linking to TShock developers. Obviously this is all vague and stuff, but I want to credit active or semi-active TShock developers in some way that might reward them. Even $1-$5 is a nice thankyou if someone implements a feature you like. I think this is really reasonable, but the "button problem" is confusing enough that I want to ask for opinions before changing. I briefly tried to change it before, but I undid the change rapidly out of fear for how to actually manage it.

Thoughts?

**Comment by Kojirremer** [comment]
Thanks a lot. That was pretty quick.

**Comment by ForsakenIdol** [comment]
As a first time Terraria server maintainer, your releases have worked flawlessly and are super intuitive to use. Kudos to you!

**Comment by zdridox** [comment]
is there version for 1.4.2.3? or is it possible to downgrade terraria?

**Reply by moisterrific** [reply]
There's no official TShock release for 1.4.2.3 yet, and yes I heard it is possible though I've never tried myself.

**Reply by hakusaro** [reply]
This isn’t the right topic for that.

**Comment by Doihuynh** [comment]
Cam ơn
